.github/workflows/build-irods-clang-format-linter.yml

```diff
@@ -0,0 +1,29 @@
+name: build-irods-clang-format-linter
+on: pull_request
+jobs:
+    clang-format-linter:
+        runs-on: ubuntu-latest
+        steps:
+            - name: Checkout Repository
+              uses: actions/checkout@v3
+              with:
+                  fetch-depth: 2
+            - name: Install Prerequisites
+              run: |
+                  sudo apt-get update -qq
+                  sudo apt-get install -qq apt-transport-https ca-certificates
+                  sudo apt-get install -qq wget
+            - name: Install Clang Compiler
+              run: |
+                  wget -qO - https://unstable.irods.org/irods-unstable-signing-key.asc | sudo apt-key add -
+                  echo "deb [arch=amd64] https://unstable.irods.org/apt/ $(lsb_release -sc) main" | sudo tee /etc/apt/sources.list.d/renci-irods-unstable.list
+                  sudo apt-get update -qq
+                  sudo apt-get install -qq irods-externals-clang13.0.0-0
+            - name: Run Clang-Format
+              run: |
+                  export PATH=/opt/irods-externals/clang13.0.0-0/bin:$PATH
+                  git config --global clangFormat.binary clang-format
+                  git config --global clangFormat.style file
+                  git config --global clangFormat.extensions 'h,c,hpp,cpp,tpp'
+                  output=`git clang-format HEAD^ --diff`
```

Comment:
It looks like this is only going to run the check on the last commit against the previous commit, which in many cases will also be part of the branch for the PR.

To demonstrate this problem, you can add a commit with a formatting violation and then add another commit on top of that with no violations. I suspect the Github Action will give a green checkmark in spite of the formatting violation.

Is there a way to run this against the whole set of commits and diff it against the base commit? Is it worth adding this if we can't run it against the complete diff?

Reply:
Good catch and yes, we should be able to run this against a series of commits.

We are free to derive the commits however we want. If bash does not make this trivial, then we can use python or whatever else we feel is a good tool.

If we can't handle multiple commits, then no, we cannot support this workflow.

Reply:
perhaps the approach here... https://github.com/KhronosGroup/SPIRV-Tools/commit/235ef34571dc734fa63a27aa672244b1548e5758 ?

Reply:
I think I got it.

After some testing, it appears that `git clang-format` can be passed the name of the branch of the target repo and it will diff/format only those lines that were changed in the PR.

See example with multiple commits here: https://github.com/korydraughn/irods/pull/3

Reply:
nice.   so the output of clang format will tell you what you 'need to do' aka what you 'should have done'.

Reply:
That's correct.